geomesa-gs-security/src/main/scala/org/geomesa/gs/security/audit/ElasticRequestDataListener.scala

```diff
@@ -0,0 +1,127 @@
+/***********************************************************************
+ * Copyright (c) 2013-2020 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the GNU GENERAL PUBLIC LICENSE,
+ * Version 2 which accompanies this distribution and is available at
+ * https://opensource.org/licenses/GPL-2.0.
+ ***********************************************************************/
+
+package org.geomesa.gs.security.audit
+
+import java.lang.reflect.Type
+import java.net.{MalformedURLException, URL}
+import java.util.Date
+
+import com.google.gson._
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.http.HttpHost
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.client.{RequestOptions, RestClient, RestHighLevelClient}
+import org.elasticsearch.action.index.{IndexRequest, IndexResponse}
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.action.DocWriteResponse
+import org.geoserver.monitor.{RequestData, RequestDataListener}
+import org.opengis.geometry.BoundingBox
+
+
+
+class ElasticRequestDataListener extends RequestDataListener with LazyLogging {
+  import ElasticRequestDataListener.gson
+  val envVar = sys.env.getOrElse("ELASTICSEARCH_HOST", null)
+  //initialized variables
+  var host = ""
+  var port = 0
+  var protocol = ""
+
+  try {
+    val url = new URL(envVar)
+    host = url.getHost
+    port = url.getPort
+    protocol = url.getProtocol
+  }
+  catch {
+    case e: MalformedURLException =>
+      logger.error("Bad URL given. Could not resolve " + envVar)
+      throw new Exception("Given URL cannot be read by Java URL")
+  }
+  val client = new RestHighLevelClient(
+    RestClient.builder(
+      new HttpHost(host, port, protocol)
+    )
+  )
+  val index = sys.env.getOrElse("GEOSERVER_ES_INDEX", null)
```

Comment:
this should probably throw an exception if null

Reply:
if you include the geomesa-gs-security jar, this class will get loaded automatically due to the applicationContext bean, right? we should probably make it so that if the env vars aren't set, the class doesn't throw any errors but just disables itself. otherwise it seems likely that just dropping the gs-security jar into your geoserver will cause it to die unless you enable this plugin with the right sys props.

Reply:
per my other comment below, i'm less concerned about it blowing up geoserver if it's in its own module/jar, although it would still probably be a good idea if you have time.

Reply:
the class shouldn't load unless the profile 'elasticsearch' is set active, so unless that gets set somewhere it will just ignore the bean, but I'll put it into it's own module too.

Reply:
ah, nice, i wasn't aware of spring profiles.